libraries/BufferedBleSpp/Xamarin Forms/StartupPage.cs

```
@@ -0,0 +1,144 @@
+﻿using Plugin.BLE;
+using Plugin.BLE.Abstractions.Contracts;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace BufferedBleSppXamarin
+{
+    internal class StartupPage : ContentPage
+    {
+        const string BLE_DEVICE_NAME = "BufferedBleSppServer";
+
+
+        IBluetoothLE ble;
+        IAdapter adapter;
+        List<IDevice> deviceList = new List<IDevice>();
+
+        BufferedBleSpp spp;
+
+        ProgressBar progressBar;
+        Label messageLabel;
+        public StartupPage()
+        {
+            progressBar = new ProgressBar();
+
+            messageLabel = new Label();
+            var connectButton = new Button { Text = "Connect" };
+            connectButton.Clicked += async (s, e) => {
+                try
+                {
+                    IDevice device = deviceList.FirstOrDefault(o => o.Name == BLE_DEVICE_NAME);
+                    if (device == null)
+                    {
+                        messageLabel.Text = $"Device {device.Name} has not been detected during scan";
+                        return;
+                    }
+                    spp = new BufferedBleSpp(adapter, device);
+                    await spp.Connect();
+                    messageLabel.Text = $"Device [{device.Name}] connected";
+                }
+                catch (Exception ex)

```

any reason to catch all exceptions? A subset should be enough?